src/Component.php

```diff
@@ -178,23 +181,27 @@ protected function getVariantDefaultVariables(string $modifier, array $defined_v
   }
 
   /**
-   * Returns the component's pathname, name, and a list of variants.
+   * Returns the component's compound name, name, and a list of variants.
    *
    * @param string $compound_name
    *   A compound name including the component and optionally variants, delimited
    *   by double-hyphens (`--`).
    *
    * @return array
    *   An array with three elements:
-   *   1. the component's pathname
+   *   1. the component compound name
```

Comment:
It does actually return the pathname which is then assigned to the property `$this->pathname` – why did you change this?

Reply:
Yes and no. The compound name consists of e.g. `@molecules/teaser/teaser--compact.twig` (but should maybe just `teaser--compact.twig`). So if we consider this full string as the path, we should rename the parameter name as well, and get rid of the confusing `$compound_name` variable name.

Reply:
The passed argument is the originally passed pathname (user input) from the calling code / twig template.  It might be a virtual name containing one or more variant suffixes.

The returned pathname is the validated canonical pathname of the component.  It is assigned to the property and later used for reading the component's definition/configuration file.